website/content/docs/agent-and-proxy/agent/exec.mdx

```diff
@@ -0,0 +1,47 @@
+---
+layout: docs
+page_title: Vault Agent Exec
+description: >-
+  Vault Agent's Exec functionality allows Vault secrets to be injected into a
+  process via environment variables using Consul Template markup.
+---
+
+# Vault Agent Exec
+
+Vault Agent's Exec functionality allows Vault secrets to be injected into a
+process via environment variables using [Consul Template markup][consul-templating-language].
+
+<Tip title="Kubernetes">
+  If you are running your applications in a Kubernetes cluster, we recommend evaluating the
+  [Vault Secrets Operator](/vault/docs/platform/k8s/vso)
+</Tip>
+
+<Warning>
+  Vault Agent Exec is in public beta. <br />
+  Please provide your feedback by opening a GitHub issue [here](https://github.com/hashicorp/vault/issues)
+</Warning>
+
+
+## Functionality
+
+The `exec` stanza allows you to configure a child process for Vault Agent to run. Secrets are injected into the process
+via environment variables configured from `env_template` blocks (see [Template Configurations](template-config) for more info).
+When you start Vault Agent with `exec` enabled, it will wait until each environment variable template is rendered at
+least once before starting the process.
+Vault Agent will restart the process when an update to a injected secret is detected. The frequency secrets are
+checked is controlled by the `static_secret_render_interval` configuration in the `template_config` stanza.
+
+## Configuration
+
+The top level `exec` block has the following configuration entries:
```

Comment:
It might make sense to copy the config parameters specific to `env_template` here or explicitly reference them with a huge link (since people will be looking for them).